### DISCUSSION TITLE
link to externally generated static content in the same site

### DISCUSSION THREAD

**Discussion by ben-marengo-msmg** [squidfunk/mkdocs-material#4860] [Q&A]
i have database schema documentation that is generated by another tool. i'd like to link to it using mkdocs.

**consider the following setup:**

docs/database-docs.md
```markdown
# database docs

[all schema docs](../externally-generated-content/databases/)
[my-db1 schema docs](../externally-generated-content/databases/my-db1.html)
[my-db2 schema docs](../externally-generated-content/databases/my-db2.html)
```

**scenario 1**
i run `mkdocs serve`
which succeeds, but, as expected, displays `WARNING  -  Documentation file 'database-docs.md' contains a link to '../externally-generated-content/databases/my-db1.html' which is not found in the documentation files.`

i then copy my `externally-generated-content` into the `site` dir

now i visit
1. http://127.0.0.1:8000/externally-generated-content/databases/ (directory of html files)
2. http://127.0.0.1:8000/externally-generated-content/databases/my-db1.html (db documentation)
3. http://127.0.0.1:8000/xyz (doesnt exist)

all show the mkdocs 404 page

**scenario 2**
i run `mkdocs build`
which succeeds, but, as expected, displays `WARNING  -  Documentation file 'database-docs.md' contains a link to '../externally-generated-content/databases/my-db1.html' which is not found in the documentation files.`

i then copy my `externally-generated-content` into the `site` dir

now i `docker run -p "8000:80" -v "${PWD}/site/:/usr/local/apache2/htdocs/" --name docs --rm httpd:2.4`

and visit the same urls
1. http://127.0.0.1:8000/externally-generated-content/databases/ (directory of html files)
2. http://127.0.0.1:8000/externally-generated-content/databases/my-db1.html (db documentation)
3. http://127.0.0.1:8000/xyz (doesnt exist)

visiting 1 shows me the static directory listing of all html files
visiting 2 shows me the database documentation html
visiting 3 shows apache 404 page

**my desired behaviour:**
1 shows me the static directory listing of all html files
2 shows me the database documentation html
3 shows mkdocs 404 page

is it possible to register static routes that are ignored by mkdocs routing in order to get my desired behaviour?

**Comment by squidfunk** [comment]
Thanks for asking. I'm not sure how to tell MkDocs to ignore those files, so it might be a good idea asking the maintainers of MkDocs by posting on their [discussion board](https://github.com/mkdocs/mkdocs/discussions). If you manage to solve the problem, let us know.
